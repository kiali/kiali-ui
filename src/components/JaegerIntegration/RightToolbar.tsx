import * as React from 'react';
import { ToolbarRightContent, Button, Icon } from 'patternfly-react';

interface RightToolbarProps {
  disabled: boolean;
  onSubmit: () => void;
}

const RightToolbar = (props: RightToolbarProps) => (
  <ToolbarRightContent>
    <Button
      bsStyle={'primary'}
      bsSize={'lg'}
      style={{ fontSize: '15px' }}
      title={'Search'}
      onClick={() => props.onSubmit()}
      disabled={props.disabled}
    >
      <Icon type="pf" name="search" /> Search
    </Button>
  </ToolbarRightContent>
);
export default RightToolbar;
