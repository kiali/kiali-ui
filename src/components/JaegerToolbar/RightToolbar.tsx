import * as React from 'react';
import { ToolbarRightContent, Button, Icon } from 'patternfly-react';

interface RightToolbarProps {
  disabled: boolean;
  graph: boolean;
  minimap: boolean;
  summary: boolean;
  onGraphClick: (state: boolean) => void;
  onSummaryClick: (state: boolean) => void;
  onMinimapClick: (state: boolean) => void;
  onSubmit: () => void;
}

export class RightToolbar extends React.PureComponent<RightToolbarProps, {}> {
  static active = { color: '#0088ce' };

  constructor(props: RightToolbarProps) {
    super(props);
  }

  render() {
    const { disabled, graph, minimap, summary, onGraphClick, onSummaryClick, onMinimapClick, onSubmit } = this.props;
    return (
      <ToolbarRightContent>
        <Button title={'Graph'} style={graph ? RightToolbar.active : undefined} onClick={() => onGraphClick(graph)}>
          <Icon type="fa" name="th" />
        </Button>
        <Button
          title={'Minimap'}
          style={minimap ? RightToolbar.active : undefined}
          onClick={() => onMinimapClick(minimap)}
        >
          <Icon type="fa" name="map" />
        </Button>
        <Button
          title={'Summary'}
          style={summary ? RightToolbar.active : undefined}
          onClick={() => onSummaryClick(summary)}
        >
          <Icon type="fa" name="info" />
        </Button>
        <Button
          bsStyle={'link'}
          title={'Search'}
          style={{ borderLeft: '1px solid #d1d1d1', marginLeft: '10px' }}
          onClick={onSubmit}
          disabled={disabled}
        >
          <Icon type="pf" name="search" />
        </Button>
      </ToolbarRightContent>
    );
  }
}

export default RightToolbar;
