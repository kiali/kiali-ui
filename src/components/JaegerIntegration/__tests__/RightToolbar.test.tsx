import * as React from 'react';
import { shallow } from 'enzyme';
import RightToolbar from '../RightToolbar';
import { shallowToJson } from 'enzyme-to-json';

describe('RightToolbar', () => {
  let wrapper, onSubmit;
  beforeEach(() => {
    onSubmit = jest.fn();
    const props = {
      disabled: false,
      onSubmit: onSubmit
    };
    wrapper = shallow(<RightToolbar {...props} />);
  });

  it('renders RightToolbar correctly', () => {
    expect(shallowToJson(wrapper)).toBeDefined();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
