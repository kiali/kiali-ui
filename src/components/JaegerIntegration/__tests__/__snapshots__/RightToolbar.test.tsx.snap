// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightToolbar renders RightToolbar correctly 1`] = `
<Button
  aria-label="SearchTraces"
  isDisabled={false}
  onClick={[Function]}
  variant="primary"
>
  <SearchIcon
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
    title={null}
  />
   Search Traces
</Button>
`;
